Cognite.Simulator.Utils/ConfigurationLibraryBase.cs

```diff
@@ -194,45 +213,166 @@ public abstract class ConfigurationLibraryBase<T, U, V> : FileLibrary<T, U>, ICo
-                        System.IO.File.ReadAllText(file.FilePath),
-                        new JsonSerializerSettings
-                        {
-                            ContractResolver = new DefaultContractResolver
-                            {
-                                NamingStrategy = new CamelCaseNamingStrategy()
-                            },
-                            Converters = new List<JsonConverter>()
-                            {
-                                new Newtonsoft.Json.Converters.StringEnumConverter()
-                            }
-                        });
-                    if (!SimulationConfigurations.ContainsKey(file.Id))
-                    {
-                        SimulationConfigurations.Add(file.Id, json);
+            // throw new Exception("not implemented");
+
+            var routinesRes = await CdfSimulatorResources.ListSimulatorRoutinesAsync(
+                new SimulatorRoutineQuery() {
+                    Filter = new SimulatorRoutineFilter() {}
+                },
+                token
+            ).ConfigureAwait(false);
+
+            var routinesMap = routinesRes.Items.ToDictionary(r => r.ExternalId, r => r);
+
+            var routineRevisionsRes = await CdfSimulatorResources.ListSimulatorRoutineRevisionsAsync(
+                new SimulatorRoutineRevisionQuery() {
+                    Filter = new SimulatorRoutineRevisionFilter() {
+                        // TODO filter by created time, simulatorExternalIds, simulatorIntegrationExternalIds
+                        // CreatedTime = new CogniteSdk.TimeRange() {  Min = _libState.DestinationExtractedRange.Last.ToUnixTimeMilliseconds() + 1 },
                     }
-                    else
-                    {
-                        SimulationConfigurations[file.Id] = json;
+                },
+                token
+            ).ConfigureAwait(false);
+
+            var simulators = _simulators.ToDictionary(s => s.Name, s => s);
+
+            // TODO: what do we do with the timerange now that we don't use FileLibrary?
+            // TODO: we need our own _libState
+            var routineRevisions = routineRevisionsRes.Items.Where(
+                r => simulators.ContainsKey(r.SimulatorExternalId)
+            ).ToList();
+
+            foreach (var routineRev in routineRevisions) {
+                if (!SimulationConfigurations.ContainsKey(routineRev.Id.ToString())) {
+                    if (routineRev.Script == null) {
+                        Logger.LogWarning("Skipping routine revision {Id} because it has no routine", routineRev.Id);
+                        continue;
+                    } else {
+                        var routineResource = routinesMap[routineRev.RoutineExternalId];
+                        // TODO: we should rather use the new type natively
+                        var simulationConfigurationWithRoutine = new SimulationConfigurationWithRoutine() {
```

Comment:
mapping the new strict type from the API directly to the current SimulationConfigurationWithRoutine